tests/core/pthread/test_pthread_dlopen.c

```diff
@@ -24,14 +24,17 @@ static void* thread_main() {
   // Spin until the main thread has loaded the side module
   while (!ready) {}
 
-  // Without this explict sync we get "invalid index into function table" below
-  _emscripten_thread_sync_code();
+  // Without this explict yield we could "invalid index into function table"
+  // below because this thread might not have loaded the side module.
+  // Uncommend the calls to printf will also, in practice, end up loading
+  // the side module because internally they may end up wait on a lock.
+  sched_yield();
 
-  printf("calling p_side_data_address=%p\n", p_side_data_address);
+  //printf("calling p_side_data_address=%p\n", p_side_data_address);
```

Comment:
so, printf should work after the yield returns? But you comment them out because in case of failure, we want the assertion to fail rather than the printf failing?

Reply:
If I remove the above sched_yield, I want the test to fail.   However, when I did that, the printf calls would trigger the code loading causing the tests to succeed after all!   

I can leave them in there without commenting them out.. but then the failure case will be harder to test.